anndata/tests/test_backed_sparse.py

```diff
@@ -149,26 +156,40 @@ def make_one_elem_mask(size: int) -> np.ndarray:
 
 # test behavior from https://github.com/scverse/anndata/pull/1233
 @pytest.mark.parametrize(
-    "make_bool_mask",
+    "make_bool_mask,should_trigger_optimization",
     [
-        make_randomized_mask,
-        make_alternating_mask,
-        make_one_group_mask,
-        make_one_elem_mask,
+        (make_randomized_mask, None),
```

Comment:
Not sure what to do about randomized masks since their behavior can't be reasoned about without calculating the mean slice length, which is somewhat circular reasoning in a unit test context that relies on that calculation being correct in the function itself.

Reply:
I think this is fine.